java-checks/src/main/java/org/sonar/java/checks/OneDeclarationPerLineCheck.java

```diff
@@ -120,20 +120,39 @@ private JavaQuickFix getQuickFixes(List<VariableTree> nodesToReport) {
 
   private JavaTextEdit getEditForVariable(VariableTree variableTree, SyntaxToken previousToken, String indentationOfLine) {
     if (",".equals(previousToken.text())) {
-      return JavaTextEdit.replaceTextSpan(textSpanBetween(previousToken, true, variableTree.simpleName(), false),
-        String.format(";\n%s%s ", indentationOfLine, modifiersAndType(variableTree)));
+      boolean isTypeFragmented = isTypeFragmented(variableTree);
+      Tree endTree = isTypeFragmented ? variableTree.type().lastToken() : variableTree.simpleName();
+      return JavaTextEdit.replaceTextSpan(
+        textSpanBetween(previousToken, true, endTree, isTypeFragmented),
+        String.format(";\n%s%s ", indentationOfLine, modifiersAndType(variableTree, isTypeFragmented)));
     } else {
       return JavaTextEdit.replaceTextSpan(textSpanBetween(previousToken, false, variableTree, false),
         String.format("\n%s", indentationOfLine));
     }
   }
 
-  private String modifiersAndType(VariableTree variableTree) {
-    ModifiersTree modifiers = variableTree.modifiers();
-    if (modifiers.isEmpty()) {
-      return QuickFixHelper.contentForTree(variableTree.type(), context);
+  private String modifiersAndType(VariableTree variableTree, boolean isTypeFragmented) {
+    StringBuilder sb = new StringBuilder();
+    if (!variableTree.modifiers().isEmpty()) {
+      sb.append(QuickFixHelper.contentOfTreeTokens(variableTree.modifiers(), context));
+      sb.append(" ");
     }
-    return QuickFixHelper.contentForRange(variableTree.modifiers().firstToken(), variableTree.type().lastToken(), context);
+    sb.append(QuickFixHelper.contentOfTreeTokens(variableTree.type(), context));
+    sb.append(" ");
+    if (isTypeFragmented) {
+      sb.append(variableTree.simpleName().name());
+    }
+    return sb.toString();
+  }
+
+  private static boolean isTypeFragmented(VariableTree variableTree) {
+    var typeTokens = ((JavaTree) variableTree.type()).allTokens();
+    for (var token : typeTokens) {
+      if (token.range().end().isAfter(variableTree.simpleName().identifierToken().range().start())) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
We can skip the loop, because: if any token in the list is placed after the variable name, then this is true also for the last token. So let's check the last token only:

```suggestion
    var lastToken = typeTokens.get(typeTokens.size()-1);
    return (lastToken.range().end().isAfter(variableTree.simpleName().identifierToken().range().start()));
```